### DISCUSSION TITLE
如何无感切换主题

### DISCUSSION THREAD

**Discussion by Half-Sugar-Code** [ant-design/ant-design#49456] [General]
### What problem does this feature solve?

- ant v5现在的定制主题真的很棒，但是现在的主题色适配方案有一个问题，在 `Next.Js`中,如果想首屏渲染的时候无感切换主题色，现在看起来不能办到。
- 原因是：假如我默认的是 `dark`模式，本地存储内有一份主题色是` light`模式，那么当进入到页面的时候会先展示为 `dark` 模式，在 `useEffect` 从本地存储内拿到主题色，再切换为 `light` 模式，这样会导致屏幕主题色会从`dark`模式闪烁一下进入到`light`模式。

### What does the proposed API look like?

我希望可以直接读取根标签上类名颜色来生成首屏的主题样式

**Comment by liixing** [comment]
https://ant-design.antgroup.com/docs/react/css-variables-cn

**Comment by arvinxx** [comment]
https://ant-design.github.io/antd-style/best-practice/fix-switch-theme-fouc
